modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoCliClientCodegen.java

```diff
@@ -0,0 +1,127 @@
+package io.swagger.codegen.languages;
+
+import io.swagger.codegen.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Map;
+
+public class GoCliClientCodegen extends PureCloudGoClientCodegen {
+    protected Logger LOGGER = LoggerFactory.getLogger(GoCliClientCodegen.class);
+
+    public GoCliClientCodegen() {
+        super();
+
+        typeMapping.put("integer", "int");
+        typeMapping.put("long", "int");
+
+        outputFolder = "generated-code/go";
+
+        embeddedTemplateDir = templateDir = "go";
+    }
+
+    @Override
+    public String getName() {
+        return "clisdkclient";
+    }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        // Wipe out all the templates so we can start with a clean slate
+        this.apiTemplateFiles.clear();
+        this.apiDocTemplateFiles.clear();
+        this.apiTestTemplateFiles.clear();
+        this.modelTemplateFiles.clear();
+        this.modelDocTemplateFiles.clear();
+        this.modelTestTemplateFiles.clear();
+        this.operationTemplateFiles.clear();
+        this.supportingFiles.clear();
+
+        if (additionalProperties.containsKey(CodegenConstants.PACKAGE_NAME)) {
+            setPackageName((String) additionalProperties.get(CodegenConstants.PACKAGE_NAME));
+        }
+        else {
+            setPackageName("swagger");
+        }
+
+        if (additionalProperties.containsKey(CodegenConstants.PACKAGE_VERSION)) {
+            setPackageVersion((String) additionalProperties.get(CodegenConstants.PACKAGE_VERSION));
+        }
+        else {
+            setPackageVersion("1.0.0");
+        }
+
+        additionalProperties.put(CodegenConstants.PACKAGE_NAME, packageName);
+        additionalProperties.put(CodegenConstants.PACKAGE_VERSION, packageVersion);
+
+        apiPackage = packageName;
+
+        supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
+        supportingFiles.add(new SupportingFile("Makefile.mustache", "", "Makefile"));
+        supportingFiles.add(new SupportingFile("gitignore.mustache", "", ".gitignore"));
+        supportingFiles.add(new SupportingFile("restclient.mustache", "/src/restclient", "restclient.go"));
+        apiTemplateFiles.put("api.mustache", ".go");
+    }
+
+    public String apiFileFolder() {
+        return (outputFolder + "/src/cmd").replace('/', File.separatorChar);
+    }
+
+    @Override
+    public String toApiFilename(String name) {
+        // replace - with _ e.g. created-at => created_at
+        name = name.replaceAll("-", "_"); // FIXME: a parameter should not be assigned. Also declare the methods parameters as 'final'.
```

Comment:
I don't understand the fixme comment. Can it be fixed now so the comment can be removed?

Reply:
Yeah that was left from the copy-paste from GoClientCodegen, it's redundant now so I'll remove it